src/firmware/boards/powerboard_esp32/PowerLoop.cpp

```diff
@@ -0,0 +1,275 @@
+
+#define EEPROM_SIZE 4
+
+SoftwareSerial SUART(2,3); //SRX = DPin-2; STX = DPin-3
+
+//Message Storage
+char INITIAL_MESSAGE;
+char MESSAGE;
+
+//Action Messages Library
+char HV_MEASUREMENT;
+char GENEVA_ENCODER;
+char GENEVA_MOTOR_ANGLE;
+char GENEVA_MOTOR_SPEED;
+char BREAKBEAM;
+char CHICKER_AUTO_KICK;
+char CHICKER_AUTO_CHIP;
+char CHICKER_KICK_DISTANCE_SPEED;
+char CHICKER_CHIP_DISTANCE_SPEED;
+char PULSE_WIDTH;
+char FLYBACK;
+
+//PINS
+const int HV_MEASUREMENT_PIN;
+const int G_ENCODER_PIN;
+const int G_MOTOR_ANGLE_PIN;
+const int G_MOTOR_SPEED_PIN;
+const int BREAKBEAM_INTER_PIN;
+const int BREAKBEAM_DIFF_PIN;
+const int CHICKER_AUTOKICK_PIN;
+const int CHICKER_AUTOCHIP_PIN;
+const int KICK_DISTANCE_SPEED_PIN;
+const int KICK_DISTANCE_SPEED_VAL_PIN;
+const int CHIP_DISTANCE_SPEED_PIN;
+const int CHIP_DISTANCE_SPEED_VAL_PIN;
+const int PULSE_WIDTH_PIN;
+const int FLYBACK_FAULT_PIN;
+
+//Flash Address
+int HV_MEASUREMENT_ADDRESS;
+int G_ENCODER_ADDRESS;
+int G_MOTOR_ANGLE_ADDRESS;
+int G_MOTOR_SPEED_ADDRESS;
+int BREAKBEAM_INTER_ADDRESS;
+int BREAKBEAM_DIFF_ADDRESS;
+int CHICKER_AUTOKICK_ADDRESS;
+int CHICKER_AUTOCHIP_ADDRESS;
+int KICK_DISTANCE_SPEED_ADDRESS;
+int CHIP_DISTANCE_SPEED_ADDRESS;
+int PULSE_WIDTH_ADDRESS;
+int FLYBACK_FAULT_ADDRESS;
+
+//Others
+int COUNTER;
+float checker;
+
+int hv_data;
+int g_encoder_angle;
+float g_motor_angle;
+float g_motor_speed;
+int breakbeam_diff;
+int breakbeam_inter;
+int autokick;
+int autochip;
+float kick_distance_speed;
+float chip_distance_speed;
+int flyback_fault;
+
+
+
+void setup(){
+
+    //Initializing serial ports 
+    Serial.begin(9600);
+    SUART.begin(9600);
+
+    //Initializing GPIO pins
+    pinMode(HV_MEASUREMENT_PIN, INPUT);
+    pinMode(G_ENCODER_PIN, INPUT);
+    pinMode(G_MOTOR_ANGLE_PIN, INPUT);
+    pinMode(G_MOTOR_SPEED_PIN, INPUT);
+    pinMode(BREAKBEAM_DIFF_PIN, INPUT);
+    pinMode(BREAKBEAM_INTER_PIN, INPUT);
+    pinMode(CHICKER_AUTOCHIP_PIN, INPUT);
+    pinMode(CHICKER_AUTOKICK_PIN, INPUT);
+    pinMode(KICK_DISTANCE_SPEED_PIN, INPUT);
+    pinMode(CHIP_DISTANCE_SPEED_PIN, INPUT);
+    pinMode(PULSE_WIDTH_PIN, INPUT);
+    pinMode(FLYBACK_FAULT_PIN, INPUT);
+
+    
+    // initialize EEPROM with predefined size
+    for (int i = 0 ; i < EEPROM.length() ; i++) {
+      EEPROM.write(i, 0);
+    }
+
+}
+
+
+void loop(){
+    INITIAL_MESSAGE = SUART.available(); //Checks if initial message is avaliable (If a message is recieved)
+
+    if (INITIAL_MESSAGE != 0) //If message avaliable, continue
+    {
+        
+        MESSAGE = SUART.read(); //Store entire message
+
+        if (MESSAGE == HV_MEASUREMENT){
+
+            //Action
+            for(COUNTER = 0; COUNTER < 5; COUNTER++){  //Total 2.5s of reading in 0.5 intervals
+
+                hv_data = analogRead(HV_MEASUREMENT_PIN);
+
+                Serial.println(hv_data);
+
+                delay(500); //wait 0.5s
+             
+            }

```

Comment:
when we are taking this HV measurement, all the other operations are going to stall

is there any reason why we need 5 measurements rather than just 1 and finishing up immediately?